### PR TITLE
Add exercise page

### PR DESCRIPTION
**Pull request by RetiredHedgehog**
изображение

### CODE REVIEW

**Review by inanimate631 — APPROVED** [review]

File: backend/authController.js [hunk 1]
```diff
@@ -132,25 +132,48 @@ class authController {
...
       options
     );
     const exercise = await exercisePromise.json();
-    const exerciseObj = exercise[0];
-    const videoId = workoutHelpers.workoutHelpers.createVideoId(
-      exerciseObj['Youtube link']
-    );
+    const exerciseObj = exercise[0] || {};
+
+    if (!(!!exerciseObj?.['Name'] || !!descriptionApiResponse?.['name'])) {
```
Comment (JaneEllison) [inline backend/authController.js:143]: !(!!(!!!(!!(!(!(!)
Could you simplify this piece of code and make it more readble?

File: backend/authController.js [hunk 2]
```diff
@@ -132,25 +132,48 @@ class authController {
...
+      res.status(404).json({});
+      return;
+    }
+
+    const getImg = (exerciseObj) => {
+      if (!!exerciseObj?.['Youtube link']) {
```
Comment (JaneEllison) [inline backend/authController.js:149]: Why do you need `!!` operator in such conditions?

File: frontend/src/Pages/exercise/exerciseTemplate.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <div class="exercise__top">
+      <iframe class="exercise__video" src=https://www.youtube.com/embed/dQw4w9WgXcQ?autoplay=0&modestbranding=1>
+      </iframe>
+      <div class="exercise__info-wrapper">
+        <div class="exercise__info">
+          <h2 class="exercise__title">
+          </h2>
+          <div class="exercise__info-list">
+            <h2>
+            </h2>
```
Comment (JaneEllison) [inline frontend/src/Pages/exercise/exerciseTemplate.ts:15]: Do you need this empty h2 element?

File: frontend/src/Pages/exercise/exerciseTemplate.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+          </div>
+        </div>
+        <button class="exercise__btn-add-to-training btn">Add to training</button>
```
Comment (JaneEllison) [inline frontend/src/Pages/exercise/exerciseTemplate.ts:18]: What about internationalization?

File: frontend/src/Pages/exercise/functions/fillExerciseFields.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+export default async function fillExerciseFields(
+  data: exerciseRespnose
+): Promise<void> {
+  const exerciseTitle = document.getElementsByClassName(
+    'exercise__title'
+  )[0] as HTMLElement;
+
+  if (exerciseTitle) {
+    exerciseTitle.innerText =
+      data.name || `Something went wrong! Couldn't fetch exercise's name!`;
```
Comment (JaneEllison) [inline frontend/src/Pages/exercise/functions/fillExerciseFields.ts:32]: Could it be a separate variable?
like
DEFAULT_EXERCISE_NAME

File: frontend/src/Pages/exercise/functions/fillExerciseFields.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  const exerciseDescription = document.getElementsByClassName(
+    'exercise__description'
+  )[0] as HTMLElement;
+
+  if (exerciseDescription) {
+    exerciseDescription.innerText =
+      data.description ||
+      `Something went wrong! Couldn't fetch exercise's description!`;
```
Comment (JaneEllison) [inline frontend/src/Pages/exercise/functions/fillExerciseFields.ts:42]: Could it be a separate variable?

**Review by JaneEllison — APPROVED** [review]

**Review by Doroshkeviches — APPROVED** [review]

File: backend/authController.js (hunk 2 above)
Comment (RetiredHedgehog) [reply backend/authController.js:149]: To be honest, I don't remember what my reasoning behind this was xd

**Pull request merged**